verilog/tools/kythe/kythe_facts_extractor.h

```diff
@@ -257,26 +198,11 @@ class KytheFactsExtractor {
 
   // Keeps track of scopes and definitions inside the scopes of ancestors as
   // the visitor traverses the facts tree.
-  ScopeContext vertical_scope_context_;
-
-  // Saved packages signatures alongside with their inner members.
-  // This is used for resolving references to some variables after using
-  // import pkg::*. e.g package pkg1;
-  //   function my_fun(); endfunction
-  //   class my_class; endclass
-  // endpackage
-  //
-  // package pkg2;
-  //   function my_fun(); endfunction
-  //   class my_class; endclass
-  // endpackage
-  //
-  // Creates the following:
-  // {
-  //   "pkg1": ["my_fun", "my_class"],
-  //   "pkg2": ["my_fun", "my_class"]
-  // }
-  std::map<std::string, std::vector<VName>> scope_context_;
+  VerticalScopeResolver vertical_scope_resolver_;
+
+  // Keeps track and saves the explored scopes with a <key, value> and maps
+  // every signature to its scope.
+  FlattenedScopeResolver flattened_scope_resolver_;
```

Comment:
Can these two scope resolvers be merged into one?
It's not intuitive how to use them.

Reply:
I am considering this and will follow in a coming PR in case a better implementation is found.